### ISSUE TITLE
Add Mongo db telemetry through event subscription

### ISSUE THREAD

**Issue by vhatsura** [elastic/apm-agent-dotnet#401]
Created as separate issue from https://github.com/elastic/apm-agent-dotnet/issues/396#issuecomment-513482890 for better visibility.

As C# mongo db driver doesn't support `DianosticSource`, telemetry can be added through [events subscription](https://www.mattburkedev.com/logging-queries-from-mongodb-c-number-driver/).

**Comment by vhatsura** [comment]
At the moment, mongo db telemetry is placed in external repository [/vhatsura/elastic-apm-mongo](https://github.com/vhatsura/elastic-apm-mongo).
It's working approach to extend `Elastic.Apm` with additional abilities, however, it's bring a few issues:
1. `Elastic.Apm` prefix cannot be used for nuget package as it's reserved by elastic team. Workaround: use `ElasticApm` prefix or ask elastic team to create proper nuget feed with push rights.
2. Limited use of apm logger infrastucture (#461). Will be solved.
3. Not possible to use helpful classes: [`CompositeDisposable`](https://github.com/elastic/apm-agent-dotnet/blob/68ec732ce08af467f1fb3e823e9559cda18b3bf3/src/Elastic.Apm/ApmAgentExtensions.cs#L35),  [`DiagnosticInitializer`](https://github.com/elastic/apm-agent-dotnet/blob/master/src/Elastic.Apm/DiagnosticSource/DiagnosticInitializer.cs). Workaround: copy-paste such classes

**Comment by SergeyKleyman** [comment]
@vhatsura There are two points to consider:
1. Regarding community contributions as external extensions to the official agent's distribution in general - we are working on the best practices: which nuget namespace we recommend to use, etc.
2. Regarding MongoDB support in particular - we would really welcome a PR to add support for MongoDB as part of the agent distribution. There might have been some misunderstanding regarding https://github.com/elastic/apm-agent-dotnet/pull/390#issuecomment-518041244 so please let me clarify it. We definitely welcome community proposals to support technologies which are not in our backlog. Each proposal is carefully considered and we might find some proposals not aligned with our current vision for the agent. In this case we recommend to develop and maintain the proposed feature as an external extension. As I mentioned above we take ability to develop external extensions seriously and we will do our best to make community members successful.

So to sum up :)  if you are interested we would really love to see a PR adding support for MongoDB.
